modules/bibdocfile/lib/bibdocfile.py

```diff
@@ -4861,6 +4871,23 @@ def open_url(url, headers=None, head_request=False):
     @type head_request: boolean
     @return: a file-like object as returned by urllib2.urlopen.
     """
+    # HACK: In order to support fetching files from Labs
+    if url.startswith('http://labs.inspirehep.net/') or url.startswith('https://labs.inspirehep.net/'):
+        user, password = _load_inspire_legacy_credentials()
+        s = requests.Session()
+        s.headers['User-Agent'] = make_user_agent_string('bibdocfile')
+        login_page = s.get("https://labs.inspirehep.net/login/?local=1")
+        soup = bs4.BeautifulSoup(login_page.text)
+        csrf_token = soup.find(id='csrf_token').get('value')
+        dummy = s.post("https://labs.inspirehep.net/login/?local=1", data={
+            'email': user,
+            'password': password,
+            'csrf_token': csrf_token,
+        })
```

Comment:
this will become quite expensive when labs is master. Couldn't you instead pickle the session at this point, so that you only have to login when the request fails?